androidApp/src/main/java/com/example/tasky/android/agenda/viewmodel/AgendaDetailsViewModel.kt

```diff
@@ -367,6 +372,12 @@ class AgendaDetailsViewModel(
                         state.copy(isEdit = false)
                     }
                 }
+
+                val requestCode = alarmRepository.getAgendaAlarm(agendaItem.id)?.requestCode
+
+                val notificationData = agendaItem.toNotificationData(requestCode)
+
+                alarmScheduler.schedule(notificationData)
```

Comment:
Personally, I would place this inside the repository to lessen the responsibility of the VM. And I’d suggest you wrap it in your `applicationScope` to ensure the alarm is scheduled successfully.